src/students/students.service.ts

```diff
@@ -19,11 +31,127 @@ export class StudentsService {
   }
 
-  getStudentById(id: string): Promise<Student> {
+  public getStudentById(id: string): Promise<Student> {
     return Student.findOne({ where: { user: { id } }, relations: { user: true, grades: true } });
   }
 
   getEmailsFromAddedStudents(addedStudents: Student[]): string[] {
     return addedStudents.map(student => student.user.email);
   }
+
+  public async getAllStudents(
+    pageOptionsDto: PageOptionsDto,
+  ): Promise<PageDto<StudentGradesAndEmpExpectationsResponse>> {
+    const queryBuilder = this.getAllStudentsQuery(pageOptionsDto);
+    const itemCount = await queryBuilder.getCount();
+    const { entities } = await queryBuilder.getRawAndEntities();
+    const students = entities.map(entity => mapStudentsResponse(entity));
+
+    const pageMetaDto = plainToInstance(PageMetaDto, { ...pageOptionsDto, itemCount });
+
+    return new PageDto(students, pageMetaDto);
+  }
+
+  // eslint-disable-next-line max-lines-per-function
+  private getAllStudentsQuery(
+    pageOptionsDto: PageOptionsDto,
+  ): SelectQueryBuilder<StudentProfileAndGrades> {
+    return this.dataSource
+      .createQueryBuilder()
+      .from(Student, 'student')
+      .leftJoinAndSelect('student.grades', 'grades')
+      .innerJoinAndSelect('student.profile', 'profile')
+      .skip(pageOptionsDto.skip)
+      .take(pageOptionsDto.take)
+      .select([
+        'student.id',
+        'student.createdAt',
+        'grades.courseCompletion',
+        'grades.courseEngagement',
+        'grades.projectDegree',
+        'grades.teamProjectDegree',
+        'profile.expectedTypeWork',
+        'profile.targetWorkCity',
+        'profile.expectedContractType',
+        'profile.expectedSalary',
+        'profile.canTakeApprenticeship',
+        'profile.monthsOfCommercialExp',
+      ])
+      .orderBy(
+        'student.createdAt',
+        pageOptionsDto.order,
+      ) as unknown as SelectQueryBuilder<StudentProfileAndGrades>;
+  }
+
+  public async getOneStudent(studentId: string): Promise<StudentResponse> {
+    const queryBuilder = await this.getOneStudentQuery(studentId);
+    Logger.log({ queryBuilder });
+    const student = mapGetOneStudentProfileResponse(queryBuilder);
+
+    return student;
+  }
+
+  // eslint-disable-next-line max-lines-per-function
+  private getOneStudentQuery(studentId: string): Promise<StudentResponse> {
+    return (
+      this.dataSource
+        .createQueryBuilder()
+        .from(Student, 'student')
+        .leftJoinAndSelect('student.user', 'user')
+        .leftJoinAndSelect('student.grades', 'grades')
+        .leftJoinAndSelect('student.profile', 'profile')
+        .where('student.id = :id', { id: studentId })
+        .select([
+          'student.id',
+          'student.createdAt',
+          'student.updatedAt',
+          'user.email',
+          'profile',
+          'grades',
+        ])
+        .getOne()
+        // eslint-disable-next-line max-lines-per-function
+        .then(result => {
+          return {
```

Comment:
Czemu tego nie przeniosłeś do `mappers.response.ts`?
a w ogóle czy to jest potrzebne? bo widzę, że masz już podłączoną metodę mapującą, więc robisz chyba 2 razy to samo